up42/workflow.py

```diff
@@ -468,6 +470,82 @@ def _helper_run_job(
             job.track_status()
         return job
 
+    def _helper_run_parallel_job(
+        self,
+        input_parameters_list: List[Dict] = None,
+        max_concurrent_jobs: int = 10,
+        test_job=False,
+        name: str = None,
+    ) -> "JobCollection":
+        """
+        Helper function to create and run parallel real or test jobs.
+
+        Args:
+            input_parameters_list: List of dictionary of input parameters.
+            max_concurrent_jobs: Maximum number of parallel jobs that can be triggered.
+            test_job: If set, runs a test query (search for available imagery based on your data parameters).
+            name: The job name. Optional, by default the workflow name is assigned.
+
+        Returns:
+            The spawned real or test job object.
+        """
+        if input_parameters_list is None:
+            raise ValueError(
+                "Select the job_parameters, use workflow.construct_parallel_parameters()!"
+            )
+
+        if test_job:
+            input_parameters_list = copy.deepcopy(input_parameters_list)
+            for input_parameters in input_parameters_list:
+                input_parameters.update({"config": {"mode": "DRY_RUN"}})  # type: ignore
+                logger.info("+++++++++++++++++++++++++++++++++")
+                logger.info("Running this job as Test Query...")
+                logger.info("+++++++++++++++++++++++++++++++++")
+
+        job_list = []
+        n = 0
```

Comment:
I am not sure introducing two lists is a better solution ... But if you think is more consistent, I will do it as you suggested.

Reply:
Also are you sure the batches as you defined always cover the whole input_parameter_list? like in case if the len of my parameters_list is 42, batches would be: 41 // 11 = 3. with this 3, from number 30 till end will be missed when creating params_batch. Or am I missing sth? Maybe batches should be 42//10 = 4 and then `for i in range(batches + 1)`?

Reply:
The reason I used `while` was because I wanted to avoid this calculation...

Reply:
Generally recommended to avoid while for looping over an iterator with a finite size. More used for infinite loops with a stop condition. Think this here would work well: https://stackoverflow.com/questions/434287/what-is-the-most-pythonic-way-to-iterate-over-a-list-in-chunks

Reply:
Yeah, But still need to make sure that we are covering the whole list ..., is your suggestion doing it or am i missing sth?